src/Compilers/Core/Portable/Syntax/SyntaxNavigator.cs

```diff
@@ -166,7 +169,8 @@ internal SyntaxToken GetLastToken(SyntaxNode current, Func<SyntaxToken, bool> pr
 
                         if (child.IsNode)
                         {
-                            stack.Push(child.AsNode().ChildNodesAndTokens().Reverse().GetEnumerator());
+                            Debug.Assert(child.IsNode);
```

Comment:
This assertion seems weird since it doesn't affect flow state and it was just checked in the containing 'if'.

Reply:
Think i was just a bit `Debug.Assert` happy here. The containing `if` is enough here to jsutify the `!`.